### PR TITLE
Raise ValueError if DataFrame column length does not match data

### LINKED ISSUE #14131: [BUG] DataFrame allows Series input with len(columns) > 1
**Describe the bug**
DataFrame allows Series input with len(columns) > 1

**Steps/Code to reproduce bug**
```python
In [20]: import cudf

In [21]: ser = cudf.Series(range(10))

In [22]: cudf.DataFrame(ser, columns=[1, 2])
Out[22]: 
Empty DataFrame
Columns: [1, 2]
Index: []
```

**Expected behavior**
ValueError

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
closes #14131

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by davidwendt** [comment]
Is the PR title missing a `not` ?

**Comment by mroeschke** [comment]
> Is the PR title missing a `not` ?

Yes, thanks!

**Comment by mroeschke** [comment]
cc @galipremsagar if you have a moment to review

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -645,7 +645,15 @@ def __init__(
         elif isinstance(data, (cudf.Series, pd.Series)):
             if isinstance(data, pd.Series):
                 data = cudf.Series.from_pandas(data, nan_as_null=nan_as_null)
-
+            if (
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:648]: I guess we shouldn't be raising no?:
```python
In [1]: import pandas as pd

In [2]: s = pd.Series([1, 2, 3], name=2)

In [3]: pd.DataFrame(s)
Out[3]: 
   2
0  1
1  2
2  3

In [4]: pd.DataFrame(s, columns=[])
Out[4]: 
Empty DataFrame
Columns: []
Index: []

In [5]: pd.DataFrame(s, columns=[2])
Out[5]: 
   2
0  1
1  2
2  3

In [6]: pd.DataFrame(s, columns=[2, 1])
Out[6]: 
   2    1
0  1  NaN
1  2  NaN
2  3  NaN

In [7]: pd.DataFrame(s, columns=[1])
Out[7]: 
Empty DataFrame
Columns: [1]
Index: []
```

**Review by galipremsagar — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/core/dataframe.py:648]: Yeah I guess it's this very narrow use case that should only raise. (I'm not sure of the history why the other cases are allowed)

**Pull request closed without merging**